### PR TITLE
Add check for in-progress same type purchase

### PR DESCRIPTION
**Pull request by ErikPeterson**

### CODE REVIEW

File: spec/acceptance/cart_purchases_controller_spec.rb [hunk 1]
```diff
@@ -129,6 +129,31 @@
...
+      }
+      other_cart = create(:cart, origin: 'subscription', user_id: cart.user_id)
+      create(:cart_purchase, cart_id: other_cart.id)
+
+      cart.update(origin: 'subscription')
+      expected_error = { errors: [{title: "Bad Request", detail: "Cannot purchase while another purchase is in progress", code: "400", status: "400"}]}
+      log_in_user(cart.user_id)
+
+      do_request params
+      p response_json
```
Comment (erinzobitz) [inline spec/acceptance/cart_purchases_controller_spec.rb]: did you mean to leave this `p` here?

Comment (ErikPeterson) [reply spec/acceptance/cart_purchases_controller_spec.rb]: No! Thank you

File: app/processors/shopping/cart_purchase_processor.rb [hunk 2]
```diff
@@ -14,6 +14,10 @@ def create_resource
         raise Shopping::NotAuthorizedError.new('not authorized', resource_klass: Shopping::CartPurchase)
       end
 
+      if (other_cart = Cart.where(origin: cart.origin, failed_at: nil, purchased_at: nil, user_id: cart.user_id).where.not(id: cart.id).first) && other_cart.cart_purchases.incomplete.exists?
```
Comment (erinzobitz) [inline app/processors/shopping/cart_purchase_processor.rb]: possible to declare `other_cart` above the `if` for readability and then just checkout `other_cart && ther_cart.cart_purchases.incomplete.exists?` ?

Comment (ErikPeterson) [reply app/processors/shopping/cart_purchase_processor.rb]: For sure

**Pull request merged**